holidays/countries/cocos_islands.py

```diff
@@ -0,0 +1,265 @@
+
+    def __init__(self, islamic_show_estimated: bool = True, *args, **kwargs):
+        """
+        Args:
+            islamic_show_estimated:
+                Whether to add "estimated" label to Islamic holidays name
+                if holiday date is estimated.
+        """
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        StaticHolidays.__init__(self, CocosIslandsStaticHolidays)
+        IslamicHolidays.__init__(
+            self, cls=CocosIslandsIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_observed(self._add_new_years_day(tr("New Year's Day")))
+
+        # Australia Day.
+        self._add_observed(self._add_holiday_jan_26(tr("Australia Day")))
+
+        act_of_self_determination_dates = {
+            2007: (APR, 5),
+            2019: (APR, 10),
+        }
+        # Act of Self Determination Day.
+        if self._year in act_of_self_determination_dates:
+            self._add_holiday(
+                tr("Act of Self Determination Day"),
+                act_of_self_determination_dates[self._year],
+            )
+        else:
+            self._add_observed(self._add_holiday_apr_6(tr("Act of Self Determination Day")))
+
+        # Good Friday.
+        self._add_good_friday(tr("Good Friday"))
+
+        # Easter Monday.
+        self._add_easter_monday(tr("Easter Monday"))
+
+        # ANZAC Day.
+        self._add_observed(self._add_holiday_apr_25(tr("ANZAC Day")))
+
+        queens_kings_birthday_dates = {
+            2021: (JUN, 7),
+            2022: (JUN, 6),
+            2023: (JUN, 6),
+            2024: (JUN, 6),
+        }
+        name = (
+            # King's Birthday.
+            tr("King's Birthday")
+            if self._year >= 2023
+            # Queen's Birthday.
+            else tr("Queen's Birthday")
+        )
+        if self._year in queens_kings_birthday_dates:
+            self._add_holiday(name, queens_kings_birthday_dates[self._year])
+        else:
+            self._add_holiday_2nd_mon_of_jun(name)
+
+        # Placed before Christmas Day for proper observed calculation.
+        self._add_observed(
+            # Boxing Day.
+            self._add_christmas_day_two(tr("Boxing Day")),
+            rule=SAT_SUN_TO_NEXT_MON_TUE + MON_TO_NEXT_TUE,
+        )
+
+        # Christmas Day.
+        self._add_observed(self._add_christmas_day(tr("Christmas Day")))
+
+        # Islamic holidays.
+
+        if self._year <= 2019:
+            # Islamic New Year.
+            for dt in self._add_islamic_new_year_day(tr("Islamic New Year")):
+                self._add_observed(dt)
+
+        # Prophet's Birthday.
+        for dt in self._add_mawlid_day(tr("Prophet's Birthday")):
+            self._add_observed(dt, tr("Prophet's Birthday"))
```

Comment:
```suggestion
            self._add_observed(dt)
```
Why not fix all similar cases at once?